docs/articles/documentation/using-testcafe/common-concepts/inject-scripts-into-tested-pages.md

```diff
@@ -0,0 +1,357 @@
+
+* [path to a JavaScript file](#inject-a-javascript-file)
+* [module name](#inject-a-module)
+* [script code](#inject-script-code)
+
+You can also [inject scripts into specific pages](#provide-scripts-for-specific-pages).
+
+> Note that the API methods and configuration option support [multiple arguments](#specify-multiple-scripts).
+
+### Inject a JavaScript File
+
+Specify the JavaScript file path to inject the entire content of this file into the tested pages. You can pass a string or an object with the `path` property.
+
+*CLI*
+
+```text
+testcafe <browser> <tests> --cs <path>
+```
+
+*Programming interface*
+
+```text
+runner.clientScripts( path | { path } )
+```
+
+*Configuration file*
+
+```text
+{
+    "clientScripts": "path" | { path }
+}
+```
+
+*Test API*
+
+```text
+fixture.clientScripts( path | { path } )
+test.clientScripts( path | { path } )
+```
+
+Argument  | Type   | Description
+--------- | ------ | ---------------------------------------------------------------------------
+`path`    | String | The path to the JavaScript file whose content should be injected.
+
+#### Relative Paths
+
+Relative paths resolve from the *current working directory* when you inject scripts in:
+
+* [command line interface](../command-line-interface.md#--cs-pathpath2---client-scripts-pathpath2)
+* [programming interface](../programming-interface/runner.md#clientscripts)
+* [configuration file](../configuration-file.md#clientscripts)
+
+When you use [test API methods](../../test-api/test-code-structure.md#inject-scripts-into-tested-pages), relative paths resolve from the *test file location*.
+
+**Examples**
+
+```sh
+testcafe chrome my-tests --cs assets/jquery.js
+```
+
+```js
+runner.clientScripts('assets/jquery.js');
+// or
+runner.clientScripts({ path: 'assets/jquery.js' });
+```
+
+```js
+{
+    "clientScripts": "assets/jquery.js",
+    // or
+    "clientScripts": { "path": "assets/jquery.js" }
+}
+```
+
+> You cannot combine the `path`, [module](#inject-a-module) and [content](#inject-script-code) properties.
+
+### Inject a Module
+
+Specify the Node.js module's name to inject its content into the tested pages. Use a string or an object with the `module` property.
+
+*CLI*
+
+```text
+testcafe <browser> <tests> --cs <module>
+```
+
+*Programming interface*
+
+```text
+runner.clientScripts( module | { module } )
+```
+
+*Configuration file*
+
+```text
+{
+    "clientScripts": "module" | { module }
+}
+```
+
+*Test API*
+
+```text
+fixture.clientScripts( module | { module } )
+test.clientScripts( module | { module } )
+```
+
+Argument  | Type   | Description
+--------- | ------ | ----------------
+`module`  | String | The module name.
+
+TestCafe searches for the module's entry point with Node.js mechanisms and injects its content into the tested page.
+
+> The module must be executable in the browser. For instance, you can use [UMD](https://github.com/umdjs/umd) or [ES Harmony](http://wiki.ecmascript.org/doku.php?id=harmony:modules) modules.
```

Comment:
Should we add this remark too?

`A lot of modules from npm registry has UMD format. It means you can easily use them in tests.`

Reply:
ES Harmony modules is a legacy name for what we know as ES6 modules. I suggest not to mention this format here. In contrast to UMD modules, ES6 modules rarely contain all required dependencies. So it's easier to say that we don't support them. 

Dependencies is also an important aspect to be mentioned here. We won't be able to load even a UMD module if its dependencies are not available as global variables.

Reply:
> A lot of modules from npm registry has UMD format.

It's a judgement...

Did I get it right that there's no way to say that a particular module supports UMD unless it's indicated in its docs?

I mean, there's little practical use in saying that `a lot of modules support UMD` - everyone is interested in modules they need. One could possibly use this kind of words to promote a feature on a landing page, but here it's not necessary.

Reply:
Ok